odoo/tools/misc.py

```diff
@@ -1505,10 +1505,14 @@ def format_datetime(
     lang = get_lang(env, lang_code)
 
     locale = babel_locale_parse(lang.code or lang_code)  # lang can be inactive, so `lang`is empty
-    if not dt_format:
+    if not dt_format or dt_format == 'medium':
         date_format = posix_to_ldml(lang.date_format, locale=locale)
         time_format = posix_to_ldml(lang.time_format, locale=locale)
         dt_format = '%s %s' % (date_format, time_format)
+    elif dt_format == 'short':
+        short_date_format = babel.dates.format_date(value, format='short', locale=locale)
```

Comment:
Why using babel instead of the given `lang` format?
```suggestion
        short_date_format = posix_to_ldml(lang.date_format, locale=locale)
```